src/Microsoft.AspNetCore.Razor.Language/CodeGeneration/DefaultDocumentWriter.cs

```diff
@@ -69,7 +69,18 @@ public void RenderChildren(RazorIRNode node)
 
             public override void VisitDocument(DocumentIRNode node)
             {
+                if (Context.Options.DesignTimeMode)
+                {
+                    Context.Writer.WriteLine("// <auto-generated/>");
```

Comment:
Does not having this cause issues at runtime for instance if you have `<GenerateDocumentationFile>true</GenerateDocumentationFile>` on and `WarningsAsErrors` enabled? If so we should just generate these all the time. For comparison, here's the blob of text that System.Web generates at the top of generated files:

```C#
//------------------------------------------------------------------------------
// <auto-generated>
//     This code was generated by a tool.
//     Runtime Version:4.0.30319.42000
//
//     Changes to this file may cause incorrect behavior and will be lost if
//     the code is regenerated.
// </auto-generated>
//------------------------------------------------------------------------------

[assembly: System.CodeDom.Compiler.GeneratedCodeAttribute("ASP.NET", "4.0.30319.42000")]
[assembly: System.Security.SecurityRulesAttribute(System.Security.SecurityRuleSet.Level2)]
[assembly: System.Runtime.Versioning.TargetFrameworkAttribute(".NETFramework,Version=v4.6.1")]

```